packages/browser-wallet/src/popup/popupX/pages/Onboarding/Welcome.tsx

```diff
@@ -16,9 +21,25 @@ import Page from '@popup/popupX/shared/Page';
 import { useAsyncMemo } from 'wallet-common-helpers';
 import { getTermsAndConditionsConfig } from '@shared/utils/network-helpers';
 import { UiStyle } from '@shared/storage/types';
+import { Checkbox } from '@popup/popupX/shared/Form/Checkbox';
+import appTracker from '@shared/analytics';
 
 const bg = document.getElementsByClassName('bg').item(0);
 
+function createAnalyticsTrackingObject(checkboxTracking: boolean) {
+    const generateClientId = () => {
+        const getRandomValue = () => Math.floor(Math.random() * 1e10);
+        return `${getRandomValue()}.${getRandomValue()}`;
+    };
+
+    return {
+        accepted: checkboxTracking,
+        clientId: generateClientId(),
+        userId: `user_${Date.now()}`,
+        sessionId: Math.floor(Date.now() / 1000),
+    };
+}
```

Comment:
This might be a nitpick, but I feel like this function should be part of the `app-tracker` module. Would also be nice to explicitly state the return type of the function to get compile-time checks.